packages/store/addon/-private/ts-interfaces/record-data.ts

```diff
@@ -13,6 +13,11 @@ export interface ChangedAttributesHash {
 }
 
 export interface RecordData {
+  id: string | null;
+  clientId: string | null;
+  modelName: string;
```

Comment:
these properties and the getResourceIdentifier method aren't part of RecordData (only our own implementation). We can get the identifier from internal model for now for places we need it, later the constructor for references should receive the identifier directly.